testing/test_runner.py

```diff
@@ -1094,3 +1094,20 @@ def func() -> None:
     with pytest.raises(TypeError) as excinfo:
         OutcomeException(func)  # type: ignore
     assert str(excinfo.value) == expected
+
+
+def test_pytest_version(pytester: Pytester, monkeypatch: MonkeyPatch) -> None:
+    pytester.makepyfile(
+        """
+        import pytest
+        import os
+
+
+        def test():
+            assert os.environ.get("PYTEST_VERSION") == pytest.__version__
+    """
+    )
+    result = pytester.runpytest_inprocess()
+    # assert result.ret == 0
+    assert result.ret == ExitCode.OK
+    assert "PYTEST_VERSION" not in os.environ
```

Comment:
Just a comment:

As is, running `pytester.runpytest` will remove the `PYTEST_VERSION` from the current environment... not sure there's anything to be done here, and should really only be something to worry about when using `pytester`, probably is fine.

Reply:
I added it to just make sure its removed properly, as that ut tests if its added

Reply:
Yes yes, the test is correct. I meant that `PYTEST_VERSION` will be removed after every call to `pytester.runpytest()`.

Hmm perhaps we can make it more robust though.

Reply:
I am not sure how does this help, are we trying to make sure if someone has been using `PYTEST_VERSION` for something else, we will retrieve its value after pytest run

Reply:
I am not sure how does this help, are we trying to make sure if someone has been using `PYTEST_VERSION` for something else, we will retrieve its value and set it `PYTEST_VERSION` to after pytest run

Reply:
I am not sure how does this help, are we trying to make sure if someone has been using `PYTEST_VERSION` for something else, we will retrieve its value and set it to `PYTEST_VERSION` after pytest run

Reply:
It is part of the principle of "cleaning up after ourselves": if the calling process has set `PYTEST_VERSION` for any reason, we should restore it at the end of `PYTEST_VERSION`. 

It even makes sense in the case of `pytester.runpytest` for example.

Reply:
Thanks for the explanation!